packages/synapse-react-client/src/components/TimelinePlot/TimelinePhase.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useRef, useState } from 'react'
+import Plotly, { Layout } from 'plotly.js-basic-dist'
+import createPlotlyComponent from 'react-plotly.js/factory'
+import dayjs, { ManipulateType } from 'dayjs'
+import { Dialog } from '@mui/material'
+import {
+  ObservationCard,
+  ObservationCardSchema,
+} from '../row_renderers/ObservationCard'
+import { Row } from '@sage-bionetworks/synapse-types'
+import pluralize from 'pluralize'
+const Plot = createPlotlyComponent(Plotly)
+
+type TimepointData = {
+  timepoints: dayjs.Dayjs[]
+  hoverOverIndex: number
+}
+
+const getTimepointData = (
+  start: dayjs.Dayjs,
+  rowData: Row[],
+  schema: ObservationCardSchema,
+  hoverEventRowId?: number, //if supplied, will return the index of this row
+): TimepointData => {
+  let hoverOverIndex = -1
+  const timepoints = rowData.map((row, index) => {
+    const time = parseInt(row.values[schema.time]!)
+    const timeUnit = row.values[schema.timeUnits]
+    if (row.rowId == hoverEventRowId) {
+      hoverOverIndex = index
+    }
+    return start.add(time, timeUnit as ManipulateType)
+  })
+  return {
+    timepoints,
+    hoverOverIndex,
+  }
+}
+
+export const getMaxDate = (timepoints: dayjs.Dayjs[]) => {
+  return timepoints.reduce(
+    (maxDateItem: dayjs.Dayjs, currentDateItem: dayjs.Dayjs) => {
+      if (!maxDateItem || currentDateItem.isAfter(maxDateItem)) {
+        return currentDateItem
+      }
+      return maxDateItem
+    },
+  )
+}
+
+const getTimelineData = (timepointData: TimepointData, rowData: Row[]) => {
+  const data = timepointData.timepoints.map((timepoint, index) => {
+    const utcFormattedTimepoint = timepoint.format()
+    const isHoveredOver = index == timepointData.hoverOverIndex
+    const rowId = rowData[index].rowId
+    return {
+      x: [utcFormattedTimepoint, utcFormattedTimepoint, utcFormattedTimepoint],
+      y: [0, 0.5, 1],
+      mode: 'lines',
+      line: {
+        color: isHoveredOver ? 'black' : 'gray',
+        width: 2,
+      },
+      // Add event into in the customdata
+      customdata: [rowId, rowId, rowId],
+      // but tell Plotly that we do not want it to show a hover tooltip (we're going to handle this)
+      hoverinfo: 'none',
+    }
+  })
+  return data
+}
+
+const getLayout = (
+  start: dayjs.Dayjs,
+  end: dayjs.Dayjs,
+  color: string,
+  annotateTime?: number,
+  annotateTimeUnits?: ManipulateType,
+): Partial<Layout> => {
+  const annotations: Partial<Plotly.Annotations>[] = [
+    {
+      x: start.add(1, 'day').format(),
```

Comment:
Without this annotation, the individual phase plot shifts when we show another annotation on hover

Reply:
It's expected.  When we draw a new annotation on the plot (in the place that we want, under the x-axis) we introduce more content to fit in the same size viewport (so elements shift to accommodate)